frontend/src/components/document/Comments/CommentPopover.jsx

```diff
@@ -0,0 +1,96 @@
+import ProfileIcon from "@/assets/pfp.svg"
+import CommentInput from "@/components/document/Comments/CommentInput";
+
+import { useSlate } from "slate-react"
+import { Transforms } from "slate"
+import  {useState, useCallback, useRef, useEffect } from "react"
+import { Popover } from "antd"
+
+import { insertCommentThread, deleteMaybeComment } from "@/util/EditorCommentUtils"
+import useAddCommentThreadToState from "@/hooks/useAddCommentThreadToState";
+import { maybeCommentAtom } from "@/util/CommentRedactionState"
+import { useSetRecoilState } from "recoil";
+
+function CommentPopover ({text}) {
+
+    const inputRef = useRef(null);
+
+    useEffect(() => {
+        if (inputRef.current) {
+            inputRef.current.focus();
+        }
+    }, []);
+
+    const [comment, setComment] = useState('');
+    const setMaybeComment= useSetRecoilState(maybeCommentAtom)
+
+    const [open, setOpen] = useState(true);
+    const editor = useSlate();
+    const addComment = useAddCommentThreadToState();
+
+    const deleteComment = useCallback(() => {
+        setComment('');
+        deleteMaybeComment(editor, setMaybeComment);
+    }, [editor, setMaybeComment]);
+    
+    const handleOpenChange = (newOpen) => {
+        setOpen(newOpen);
+        if (!newOpen) {
+            deleteComment();
+            Transforms.deselect(editor);
+        }
+    };
+
+    const submitComment = useCallback(() => {
+        if (comment.length > 0) {
+          insertCommentThread(editor, addComment);
+          deleteComment();
```

Comment:
Maybe add a comment saying why this delete in necessary? Like what are we deleting from if this is to submit